### PR TITLE
Fixes #24827: When we have a user without administrator role, the menu "Utilities" exist but leads to a 404

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/24827
image
image

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala [hunk 1]
```diff
@@ -709,20 +709,13 @@ class Boot extends Loggable {
...
-          Full(RedirectWithState("/secure/index", redirection))
-        }
-      })).submenus(
+      (Menu(
+        MenuUtils.utilitiesMenu,
+        <i class="fa fa-wrench"></i> ++ <span>Utilities</span>: NodeSeq
+      ) /
+      "secure" / "utilities" / "index"
+      >> LocGroup("utilitiesGroup")
+      >> TestAccess(() => userIsAllowed("/secure/index", AuthorizationType.Administration.Read))).submenus(
```
Comment (fanf) [inline webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala:718]: doesn't that will lock-out validator/etc who are not admin_read from accessing the change request validation screen ?

Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala:718]: I think it should only be the removal of Technique Read rights, i think it was there when techique editor was below Utilities, a looooong time ago

**Comment by ElaadF** [comment]
PR updated with a new commit

**Comment by ElaadF** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala [hunk 2]
```diff
@@ -709,19 +709,19 @@ class Boot extends Loggable {
...
+      (Menu(
+        MenuUtils.utilitiesMenu,
+        <i class="fa fa-wrench"></i> ++ <span>Utilities</span>: NodeSeq
+      ) /
+      "secure" / "utilities" / "index"
+      >> LocGroup("utilitiesGroup")
+      >> TestAccess(() => {
+        userIsAllowed(
+          "/secure/index",
+          AuthorizationType.Administration.Read,
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala:721]: You still need to have workflow enabled checked. the menu should not appaear if workflow enabled is false and we don't have administrator Read. You'll still have the 404 in this case

This pr should only be removal of Technique.Read right

**Comment by ElaadF** [comment]
PR rebased

**Comment by ElaadF** [comment]
PR updated with a new commit

**Comment by ElaadF** [comment]
PR updated with a new commit

**Comment by ElaadF** [comment]
OK, squash merging this PR

**Pull request merged**